src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -70,38 +70,143 @@ public void fetchAndMerge(BibEntry entry, Field field) {
+            LOGGER.error("Error while fetching bibliographic information", exception);
+            String fetcherName = fetcher.getName();
+            if (exception instanceof FetcherClientException) {
+                dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("No data was found for the identifier"));
+            } else if (exception instanceof FetcherServerException) {
+                dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Server not available"));
             } else {
-                dialogService.notify(Localization.lang("No %0 found", field.getDisplayName()));
+                dialogService.showInformationDialogAndWait(Localization.lang("Fetching information using %0", fetcherName), Localization.lang("Error occurred %0", exception.getMessage()));
+            }
+        }
+    }
+
+    private void mergeWithoutDialog(BibEntry originalEntry, BibEntry fetchedEntry) {
+        NamedCompound ce = new NamedCompound(Localization.lang("Merge entry without user interaction"));
+
+        // Set of fields present in both the original and fetched entries
+        Set<Field> jointFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        jointFields.addAll(fetchedEntry.getFields());
+        Set<Field> originalFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        originalFields.addAll(originalEntry.getFields());
+
+        boolean edited = false;
+
+        // Compare entry type and update if needed
+        EntryType oldType = originalEntry.getType();
+        EntryType newType = fetchedEntry.getType();
+
+        if (!oldType.equals(newType)) {
+            originalEntry.setType(newType);
+            ce.addEdit(new UndoableChangeType(originalEntry, oldType, newType));
+            edited = true;
+        }
+
+        // Compare fields and set the longer value
+        for (Field field : jointFields) {
+            Optional<String> originalString = originalEntry.getField(field);
+            Optional<String> fetchedString = fetchedEntry.getField(field);
+
+            if (fetchedString.isPresent()) {
+                if (originalString.isEmpty() || fetchedString.get().length() > originalString.get().length()) {
+                    originalEntry.setField(field, fetchedString.get());
+                    ce.addEdit(new UndoableFieldChange(originalEntry, field, originalString.orElse(null), fetchedString.get()));
+                    edited = true;
+                }
             }
         }
+
+        // Remove fields not present in fetched entry, unless they are internal
+        edited = isEdited(originalEntry, ce, jointFields, originalFields, edited);
+
+        if (edited) {
+            ce.end();
+            undoManager.addEdit(ce);
+            dialogService.notify(Localization.lang("Updated entry with fetched information"));
```

Comment:
Add the citation key to this information - otherwise it is unclear which entry is used.

Use `getCitationKey().orElse(getAuthorTitleYear()`